Source/SimCacheModule/Source/Core/CacheDefinitionCollection.h

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) 2024 Steven Frost and Orion Lyau
+
+#pragma once
+
+#include <Utils/Serialisation/Serialisable.h>
+#include <Utils/Time/DateTime.h>
+
+#include <cstdint>
+#include <string>
+#include <vector>
+
+// -----------------------------------------------------------------------------
+
+typedef std::string CacheId;
+typedef std::string CacheAuthorId;
+typedef std::string Locale;
+
+// -----------------------------------------------------------------------------
+
+struct CacheMetadata
+	: public Utils::Serialisation::ISerialisable
+{
+	CacheMetadata();
+
+public: // ISerialisable
+
+	virtual bool Serialise( Utils::Serialisation::Writer& Writer ) const override;
+	virtual bool Deserialise( Utils::Serialisation::Reader& Reader ) override;
+
+public:
+
+	CacheId							Id;
+	std::vector< CacheAuthorId >	Authors;
+	Utils::DateTime					DatePlaced;
+
+	uint8_t							Difficulty;
+	std::string						SimObjectTitle;
+};
+
+// -----------------------------------------------------------------------------
+
+struct CacheText
+	: public Utils::Serialisation::ISerialisable
+{
+	CacheText();
+
+public: // ISerialisable
+
+	virtual bool Serialise( Utils::Serialisation::Writer& Writer ) const override;
+	virtual bool Deserialise( Utils::Serialisation::Reader& Reader ) override;
+
+public:
+
+	std::string Title;
+	std::string Description;
+	std::string Hint;
```

Comment:
Do we want wide strings here, since this is text that can become localized?  Though at the moment, I don't think the serialization readers/writers are set up for wide strings?  Something to note for the future?

Reply:
Yeah these will be wstring when we do a pass on localisation. There are likely wider changes required to serialisation to support that.